src/ui/tileseteditortileselector.cpp

```diff
@@ -95,36 +95,45 @@ void TilesetEditorTileSelector::updateSelectedTiles() {
 
 QList<Tile> TilesetEditorTileSelector::getSelectedTiles() {
     if (this->externalSelection) {
-        return this->externalSelectedTiles;
+        return buildSelectedTiles(this->externalSelectionWidth, this->externalSelectionHeight, this->externalSelectedTiles);
     } else {
-        QList<Tile> tiles;
-        QList<QList<Tile>> tileMatrix;
-        QList<uint16_t> selected = this->selectedTiles;
         QPoint dimensions = this->getSelectionDimensions();
-        int width = dimensions.x();
-        int height = dimensions.y();
-        for (int j = 0; j < height; j++) {
-            QList<Tile> row;
-            for (int i = 0; i < width; i++) {
-                int index = i + j * width;
-                uint16_t tile = selected.at(index);
-                if (this->xFlip)
-                    row.prepend(Tile(tile, this->xFlip, this->yFlip, this->paletteId));
-                else
-                    row.append(Tile(tile, this->xFlip, this->yFlip, this->paletteId));
-            }
-            if (this->yFlip)
-                tileMatrix.prepend(row);
+        QList<Tile> tiles;
+        for (int i = 0; i < this->selectedTiles.length(); i++) {
+            uint16_t tile = this->selectedTiles.at(i);
+            tiles.append(Tile(tile, false, false, this->paletteId));
+        }
+        return buildSelectedTiles(dimensions.x(), dimensions.y(), tiles);
+    }
+}
+
+QList<Tile> TilesetEditorTileSelector::buildSelectedTiles(int width, int height, QList<Tile> selected) {
+    QList<Tile> tiles;
+    QList<QList<Tile>> tileMatrix;
+    for (int j = 0; j < height; j++) {
+        QList<Tile> row;
+        for (int i = 0; i < width; i++) {
+            int index = i + j * width;
+            Tile tile = selected.at(index);
+            tile.xflip ^= this->xFlip;
+            tile.yflip ^= this->yFlip;
+            tile.palette = this->paletteId;
```

Comment:
This is introducing a bug where the palette id of the first-selected tile is applied to every externally-selected tile.